### PR TITLE
WIP: development docs

### PR DESCRIPTION
**Pull request by rfleschenberg**
Refs #933 

Work in progress. I would like to get some input from people who are more familiar with the project.

This is just what I deduced from `.travis.yml` so far.

Curiously, if I follow these steps to run the tests on my local system, I get some test failures that I do not see on travis: https://gist.github.com/rfleschenberg/9b3eb58154cd8456a320b6b5cb694ce8

### CODE REVIEW

**Comment by rfleschenberg** [comment]
The instructions from 302d2f3 work on my system.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+To work on pipenv itself, fork the repository and clone your fork to your local
+system.
+
+Now, install the development requirements::
+
+    cd pipenv
+    virtualenv ~/pipenv-venv  # You can use a different path if you like.
+    source ~/pipenv-venv/bin/activate
+    pip install -e . --upgrade
```
Comment (nateprewitt) [inline CONTRIBUTING.rst]: I would probably suggest using `python setup.py develop` here since it's more idiomatic.

File: MANIFEST.in [hunk 2]
```diff
@@ -1,4 +1,4 @@
-include README.rst LICENSE NOTICES HISTORY.txt pipenv/patched/safety.zip
+include README.rst CONTRIBUTING.rst LICENSE NOTICES HISTORY.txt pipenv/patched/safety.zip
```
Comment (nateprewitt) [inline MANIFEST.in]: We don't want to distribute non-essential documentation in our wheels. Let's remove this since new contributors can find it here now.

**Comment by nateprewitt** [comment]
Also if you wouldn't mind [squashing](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing) your work into a single commit, that would be greatly appreciated.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (rfleschenberg) [reply CONTRIBUTING.rst]: I was trying to stay close to what we do on travis, where we use ` pip install -e . --upgrade`.

Do you prefer to just use `setup.py develop here`, or also make the same change to `.travis.yml`?

Comment (nateprewitt) [reply CONTRIBUTING.rst]: setup.py develop is just a more universal way of installing a package locally for development and has historically been how Kenneth and I ran our dev environments.

It looks like the .travis.yml file was implemented with this syntax when we reintroduced Travis a couple months ago. @erinxocon, any particular reason we should use that syntax?

Comment (rfleschenberg) [reply CONTRIBUTING.rst]: I changed it to `python setup.py develop` for now. I guess it's easy enough to change it later if you want to.

**Comment by nateprewitt** [comment]
Wonderful, thanks for this @rfleschenberg!

**Pull request merged**